namepsace of isinf / isnan not specified

I'm getting an error during compilation for isinf/isnan, which in a few spots isn't using the std:: namespace. the compiler is looking for it in boost. 

```

Evolution/../Evolution/../Utils/../Operators/Regression/OpLog.h:49:26: error: ‘isinf’ was not declared in this scope

```

I was able to fix it by adding std:: before isinf and isnan in a few files. i'll issue a PR in a minute.

--- a/Operators/Regression/OpLog.h
+++ b/Operators/Regression/OpLog.h
@@ -46,9 +46,9 @@ public:
             v1 = exp(desired_elem[i]);
             v2 = -exp(desired_elem[i]);
             
-            if (!isinf(v1) && !isnan(v1))
+            if (!std::isinf(v1) && !std::isnan(v1))
                 res_v.push_back(v1);
-            if (!isinf(v2) && !isnan(v2))
+            if (!std::isinf(v2) && !std::isnan(v2))
                 res_v.push_back(v2);
         }
         

--- a/Semantics/SemanticLibrary.cpp
+++ b/Semantics/SemanticLibrary.cpp
@@ -143,8 +143,9 @@ bool SemanticLibrary::AddToLibrary(Node* n, Fitness& fitness, bool caching) {
     pair<double_t, double_t> mean_std;
     if (normalize_outputs) {
         mean_std = Utils::ComputeMeanStdEfficiently(out);
-        if (isinf(mean_std.first) || isnan(mean_std.first) ||
-                mean_std.second == 0 || isinf(mean_std.second) || isnan(mean_std.second))
+        if (std::isinf(mean_std.first) || std::isnan(mean_std.first) ||
+                mean_std.second == 0 || std::isinf(mean_std.second) 
+                || std::isnan(mean_std.second))
             return false;
         if (!linear_parse_library)
             out = out - mean_std.first;
@@ -296,7 +297,7 @@ std::pair< Node *, double_t> SemanticLibrary::GetClosestSubtree(const std::vecto
         }
     }
 
-    if (isinf(best_min_dist)) {
+    if (std::isinf(best_min_dist)) {
         return make_pair(best_sol, best_min_dist);
     }
 

--- a/Tests/TestsSemanticBackpropagation.h
+++ b/Tests/TestsSemanticBackpropagation.h
@@ -78,7 +78,7 @@ static void TestInversions() {
     o[0] = 0;
     res = op->Invert(d, o, 0);
 
-    assert(isnan(res[0]));
+    assert(std::isnan(res[0]));
 
     delete op;
 

--- a/Variation/SubtreeVariator.cpp
+++ b/Variation/SubtreeVariator.cpp
@@ -231,7 +231,7 @@ Node * SubtreeVariator::FindSubtreeForSemanticBackpropagationReplacement(std::ve
             for (size_t j = 0; j < desired_output[i].n_elem; j++) {
                 if (!isinf(desired_output[i][j])) {
                     vals[i] = desired_output[i][j];
-                    if (!isnan(vals[i])) {
+                    if (!std::isnan(vals[i])) {
                         vals_no_nans.push_back(vals[i]);
                         desired_vals_mean += vals[i];
                     }
@@ -255,7 +255,7 @@ Node * SubtreeVariator::FindSubtreeForSemanticBackpropagationReplacement(std::ve
             // could recompute desired_vals_mean, but that should be close to 0 anyway (current value) due to normal distributed samples above
 
         } else if (!use_lib_linear_parsing) { // compute the mean, make sure its ok
-            if (isinf(desired_vals_mean) || isnan(desired_vals_mean)) {
+            if (isinf(desired_vals_mean) || std::isnan(desired_vals_mean)) {
                 search_for_normalized_tree = false;
             }
         }
@@ -370,7 +370,7 @@ std::pair<double_t, double_t> SubtreeVariator::GenerateConstantFromDesiredOutput
         for (vec & x : desired_output) {
             double_t min_d_within_x = datum::inf;
             for (double_t v : x) {
-                if (isnan(v))
+                if (std::isnan(v))
                     continue;
                 double_t d = (v - c)*(v - c);
                 if (d < min_d_within_x) {

